test/unit-tests/sexp_tests.ml

```diff
@@ -23,13 +23,14 @@ let () =
         | Atom _ -> true
         | _      -> false
       in
-      if Usexp.Atom.is_valid s <> parser_recognizes_as_atom then begin
+      let valid_dune_atom = Usexp.Atom.is_valid_dune s in
+      if valid_dune_atom <> parser_recognizes_as_atom then begin
```

Comment:
Could you update this test so that it runs for both the jbuild and Dune syntax? For instance it would have caught the issue related to `is_valid_jbuild`